scripts/qa_crowbarsetup.sh

```diff
@@ -4786,6 +4786,20 @@ function onadmin_allow_vendor_change_at_nodes
     done
 }
 
+function crowbar_nodeupgrade_finished
+{
+    if grep current_step $upgrade_progress_file | grep -v nodes ; then
+        echo "'nodes' step finished successfuly"
+        return 0
+    fi
+
+    if grep -q failed $upgrade_progress_file ; then
+        echo "Something has failed during 'nodes' upgrade step."
```

Comment:
It's there just a few lines later, once `crowbar_nodeupgrade_finished` exits the wait cycle.